### DISCUSSION TITLE
Add to default headers in api.js

### DISCUSSION THREAD

**Discussion by larntz** [cypress-io/cypress#23970] [Questions and Help]
I would like to be able to add to these headers in [packages/server/lib/api.js](https://github.com/cypress-io/cypress/blob/238ead3c53399c022f7c99236ccc709768395a61/packages/server/lib/api.js#L61-L64). Does anyone know if there is a configuration option available to do this?

The issue we are having is that there is no user-agent header being sent to the api_url and the request is being blocked by our application firewall.

**Comment by jjhelguero** [comment]
I'm not sure of your version nor set up. Another way to do this is passing a `--config userAgent=\"User-Agent-You-Desire\"` via command line.
This is would set up to have user agent to be an iPhone.
```
cypress open --config userAgent=\"Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_1 like Mac OS X) Mobile/14E304\""
```

**Comment by larntz** [comment]
Appreciate the suggestion. I did try setting userAgent with the config option, but I will try that again and report back if it works.

Thank you!

**Reply by larntz** [reply]
This does not set the user-agent header for the initial request to the api_url. 

here is the cypress version info:

```
npx cypress version
Cypress package version: 10.8.0
Cypress binary version: 10.8.0
Electron version: 19.0.8
Bundled Node version: 16.14.2
```

and here is the run error message: 

```
npx cy2 run -b chrome --config-file integration/config/cypress.config.ts --record --key unusedKey --parallel --ci-build-id asdf-1234 --config userAgent="asdf"
[cy2] Running cypress with API URL: https://sorry-cypress-backend.example.com
cypress-grep: filtering using tag(s) "-#bug+-#flaky"
We encountered an unexpected error talking to our servers.

Because you passed the --parallel flag, this run cannot proceed because it requires a valid response from our servers.

The --ciBuildId flag you passed was: asdf-1234

The server's response was:

StatusCodeError: 403 - "<html>\r\n<head><title>403 Forbidden</title></head>\r\n<body>\r\n<center><h1>403 Forbidden</h1></center>\r\n</body>\r\n</html>\r\n"
```

That 403 response code is coming from aws wafv2 rule that blocks requests without a user-agent header. 

I also go a response from my issue on cy2 referring me back to the cypress project. They are telling me all of these network requests are coming from cypress itself. 

If anyone can help me get a user-agent header set for requests to the api_url I would greatly appreciate it!

Thanks again
